modules/ghlint-cli/src/main/kotlin/net/twisterrob/ghlint/cli/CLI.kt

```diff
@@ -45,30 +71,44 @@
 		}
 	}
 
-	// IMPORTANT: The order of properties is important, it'll define how "--help" is printed.
-
-	override val files: List<Path> by argument()
-		.path(mustExist = true, canBeDir = false)
-		.multiple(required = true)
-		.help("Workflow YML files to check for problems.")
-
-	private val inputs: InputOptions by InputOptions()
-
-	override val root: Path by option("--root")
-		.path(mustExist = true, canBeDir = true, canBeFile = false)
-		.default(Path.of("."))
-		.help("Root directory of the repository.")
-
-	override val isVerbose: Boolean by option("--verbose")
-		.flag(default = false, defaultForHelp = "off")
-		.help("Prints more information.")
-
 	init {
 		eagerOption("-v", "--version", help = "Prints the version and exits.") {
 			throw PrintMessage("GH-Lint (${commandName}) version ${GHLINT_VERSION}")
 		}
 	}
 
+
+	init {
+		eagerOption("-h", "--help", help = "Show help for the command or a specific rule ID.") {
+			@Suppress("UNCHECKED_CAST")
+			val originalArgs = currentContext.data["originalArgs"] as? List<String> ?: emptyList()
+			
+			// Find the position of --help in the arguments
+			val helpIndex = originalArgs.indexOfFirst { it == "-h" || it == "--help" }
+			
+			// Check if there's an argument after --help
+			val nextArg = if (helpIndex >= 0 && helpIndex + 1 < originalArgs.size) {
+				originalArgs[helpIndex + 1]
+			} else null
+			
+			// If there's an argument after --help that doesn't start with "-", treat it as a rule ID
```

Comment:
This argument parsing logic is fragile and could break with edge cases. What happens with '--help --verbose' or '--help -v'? Consider using Clikt's parameter transformation or a proper subcommand structure.